[3824] Dependencies dialog not visible after checking dependencies by using edit page url

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/3824

aachavarria committed:
fixing issues with z-index

aachavarria committed:
there are 2 places to open depencedies, in case 1 it should be 1041, and the case two the edit form should be 1042

rart commented on static-assets/components/cstudio-common/common-api.js:
```diff
@@ -580,10 +580,11 @@ var nodeOpen = false,
                     dialogue.render(document.body);
                     dialogue.centreY();
 
+                    console.log('este??');
```
x

rart commented on static-assets/yui/container/container.js:
```diff
@@ -5743,10 +5743,10 @@ version: 2.6.0
          */
         _onElementFocus : function(e){
-
+            var authDialog = document.getElementById('auth-monitor-dialog');
```
Camelcase ID.
Missing the AuthMonitor.tsx update that adds this ID to the dialog.

Please revert all other changes in this file, leaving only the code changes. Make sure not to format. If necessary, edit the file with something other than IntelliJ so it doesn't remove trailing whitespaces and things of that nature.

rart commented on static-assets/yui/container/container.js:
```diff
@@ -5743,10 +5743,10 @@ version: 2.6.0
             var target = Event.getTarget(e);
 
-            if (target !== this.element && !Dom.isAncestor(this.element, target) && _currentModal == this) {
+            if (target !== this.element && (!authDialog && !Dom.isAncestor(authDialog, target)) && !Dom.isAncestor(this.element, target) && _currentModal == this) {
```
Let's double check this condition...

This doesn't seem to cover when **there is an** auth dialog.

The original condition reads:
- Focused element is not the modal's element AND
- Focused element is not a child of the modal's element AND
- The modal is the topmost modal (current modal) THEN
- Set the focus to an element in this modal

We need to add:
```
AND... (
  There is no auth dialog OR (
    the focused element IS the auth dialog OR 
    the focused element IS a child of the auth dialog
  )
)
```

rart replied:
Actually, on second thought, the simple existence of an auth dialog should suffice to skip the focus trap.

rart reviewed (changes requested):
1. See code comments
2. How did you edit the minified file?

rart committed:
[3824] Dialog stacking z-index issues

https://github.com/craftercms/craftercms/issues/3824

rart committed:
[3824] Dialog stacking z-index issues

Reverts container.js whitespace/formats

https://github.com/craftercms/craftercms/issues/3824
